lib/intercom/search_collection_proxy.rb

```diff
@@ -0,0 +1,82 @@
+require "intercom/utils"
+
+module Intercom
+  class SearchCollectionProxy
+
+    attr_reader :resource_name, :resource_class
+
+    def initialize(resource_name, search_details: {}, client:)
+      @resource_name = resource_name
+      @resource_class = Utils.constantize_resource_name(resource_name)
+      @search_url = search_details[:url]
+      @search_params = search_details[:params]
+      @client = client
+    end
+
+    def each(&block)
```

Comment:
This auto-pages, right? 

Maybe it would be better to provide two methods, one to auto-page, and one to provide the functionality to not auto-page. 🤔 

WDYT?

Reply:
Yes, that's been used in `ScrollCollectionProxy` similarly. I feel it's fine to keep it like so that `each` can iterate through the full list. We can still use `.next` with a count (based on `per_page`) if no paging is required.

Reply:
Fair, that makes sense.